agent/consul/leader.go

```diff
@@ -178,7 +195,7 @@ RECONCILE:
 			if err := s.revokeLeadership(); err != nil {
 				s.logger.Printf("[ERR] consul: failed to revoke leadership: %v", err)
 			}
-			goto WAIT
+			return err
```

Comment:
This usecase #1: we couldn't establish leadership and instead of retrying it after `interval` in `WAIT`, an error is return which leads to a raft leadership transfer.